release: v0.11.0 (#1122)
rero/rero-ils: RELEASE-NOTES.rst

Release notes
=============

v0.11.0
-------

User interface
--------------

-  Updates schema of forms to use the new sorted select menu.
-  Displays new metadata fields: ``seriesStatement`` and ``partOf``:

   -  Displays fields in detailed view and in editor.
   -  Removes ``partOf`` field from brief view.
   -  Adds ``oneOf`` attribute in order to link issuance ``maintype``
      and ``subtype`` in editor.

-  Adds missing translations of item notes types.
-  Limits length of document title to 150 characters in detailed view.
   Adds a *Show more*/*Show less* link if the title is truncated.
-  Sorts pickup locations alphabetically when placing a request for an
   item.

Professional interface
~~~~~~~~~~~~~~~~~~~~~~

> -  Adds inventory list functionnality and view based on ``item``

Reviewer: ```suggestion
-  Adds inventory list functionality and view based on ``item``
```

>    resources. The can acces them using the ``Admin`` menu. This

Reviewer 2: Line 45: please change 
The can acces them using the ``Admin`` menu.
by
The librarian can acces them using the ``Admin`` menu.

>    resources. The can acces them using the ``Admin`` menu. This

Reviewer: ```suggestion
   resources. The librarian can access them using the ``Admin`` menu. This
```

>    functionnality allows the librarian to display a list of items,

Reviewer: ```suggestion
   functionality allows the librarian to display a list of items,
```

   purposes.

   -  Adds ``CSVSerializer`` to render list results to CSV.

-  Adds several improvements to the editor layout:

   -  Improves document ``JSONSchema`` form options by adding css
      classes, default values and form options to increase its
      usability.
   -  Adds borders to form groups and generally improves UI.
   -  Fixes fields to be displayed by default in editor, input sizes,
      always-hidden fields.

-  Adds a custom directive allowing to order tabs.
-  Changes the sequence of editor initialization to avoid concurrency
   problem with ``JSONSchema`` loading in BNF import editor.

Metadata
--------

-  Adds three new fields to the data model: ``seriesStatement``,
   ``partOf`` and ``issuance``.

   -  Implements transformation of these fields for ``MARC21`` and
      ``UNIMARC``.
   -  Adapts ES mappings and JSON schemas.
   -  Indexes host document title in child document’s record for search
      results relevance.

-  Adds a ``second_call_number`` field to ``item`` resource.
-  Splits document ``JSONSchema`` into smaller files, to improve
   readability. JSON references are resolved on the fly.

API
---

> -  Refactors the permission processe using a permission factory and

Reviewer: ```suggestion
-  Refactors the permission processes using a permission factory and
```

   classes for specific resources instead of all resources.

>    -  Refactors permission factories for resources: organisation,

Reviewer: There is no resource called "acquisition"

      document, acquisition, patron, item.

Documentation
-------------

-  Documents the new ``Weblate`` translation workflow.

   -  Adds a Weblate badge in the ``README.rst`` that informs about the
      completion of translations, and points to the Weblate service.
   -  Removes the check of translation message extraction in the PR
      template.

-  Improves the ``rero-ils-ui`` README and adds badges as well as
   UCLouvain in copyright declaration.

Tests
-----

-  Adds an ``id`` on all menus in order to simplify and improve Cypress
   tests.

   -  Replaces ``getId()`` by ``idAttribute`` pipe from Angular.
   -  Doesn’t hide the Debug toolbar in Cypress tests as ``FLASK_DEBUG``
      should be set to ``False`` when launching the server.
   -  Creates new ``setLanguageToEnglish`` Cypress command to set
      language to English.

>    -  Deletes all ``cy.visit()`` method and use menus to navigate in the

Reviewer: ```suggestion
   -  Deletes all ``cy.visit()`` methods and use menus to navigate in the
```

   extract messages and updates catalog.
-  Adds ``ngx-spinner`` dependency used in ``ng-core`` to
   ``rero-ils-ui``. ``ngx-spinner`` is used in version ``v8.1.0`` for
   compatibility with Angular 8.
-  The module ``rero-ils-ui`` uses the ``ng-core`` library in version
   ``v0.6.0``

Scripts
~~~~~~~

-  Fixes ``npm`` asset utils installation at ``bootstrap``: adds error
   message when npm asset utils fails and uses –force option for npm
   asset utils installation.
-  Improves ``Celery`` script option for server launching: adds new
   ``-l`` or ``--loglevel`` server script option to change Celery log
   level and adds new ``-n`` or ``--no-worker`` server script option to
   disable Celery workers.
-  Improves ``check_license`` method to include Triple-Slash directives
   for ``.js`` files, avoids checking screenshots directory in Cypress,
   and adds a triple slash directive on 2 JS files (from Cypress).

Fixed issues
------------

-  `#880`_: Reduce size of title in document detailed view
-  `#882`_: Translations of actions realised in circulation UI
-  `#883`_: Improvement needed on the request information when doing a
   checkin

> -  `#886`_: Clear the patron info on top of checkin form when quiting it

Reviewer: ```suggestion
-  `#886`_: Clear the patron info on top of checkin form when quitting it
```
